subprojects/core/src/main/java/org/gradle/execution/taskgraph/TaskInfo.java

```diff
@@ -130,11 +129,18 @@ public Throwable getTaskFailure() {
     }
 
     public boolean allDependenciesComplete() {
-        for (TaskInfo dependency : Iterables.concat(mustSuccessors, dependencySuccessors)) {
```

Comment:
This is almost certainly gonna be inlined by the JVM, so probably not going to give a speedup when unrolling manually. Even if it wasn't optimized away, I don't think it would be noticable compared to the other stuff we're doing in the task plan.